app/controllers/APIController.scala

```diff
@@ -1,30 +1,132 @@
 package controllers
 
 import services._
-import io.circe.generic.codec.DerivedAsObjectCodec.deriveCodec
 import io.circe.syntax.EncoderOps
+import io.circe._
+
 import javax.inject._
 import play.api._
 import play.api.mvc._
+import DAO.BakeryDB
+import io.circe
+import models.{ProductFromJson, ServiceStatus}
+
+import java.util.UUID
 
 @Singleton
 class APIController @Inject() (
-    val environment: Environment,
-    val controllerComponents: ControllerComponents
+    environment: Environment,
+    val controllerComponents: ControllerComponents,
+    bakeryDB: BakeryDB
 ) extends BaseController {
 
   def statusToJson: String = {
-    val status = models.Status(
+    val status = ServiceStatus(
       "pass-bakery",
       environment.mode.toString,
       APIService.getDateTime
     )
     status.asJson.spaces2
   }
 
+  def parseFromJson(request: Request[AnyContent]): Option[ProductFromJson] = {
+    val parseResult: Option[ProductFromJson] = {
+      request.body.asText match {
```

Comment:
your endpoints are still expecting a content-type of `text/plain`. How can you ask for an `application/json` content-type (which is what you want)?